lib/dynamodb.js

```diff
@@ -416,38 +349,38 @@ module.exports = function(configuration) {
 					let tasks = [];
 					for (let ndx = 0; ndx < all.length; ndx += 25) {
 						let myRecords = all.slice(ndx, ndx + 25);
-						tasks.push(function(done) {
+						tasks.push(function (done) {
 							let retry = {
 								backoff: (err) => {
 									done(null, {
-										backoff: err || "error",
-										records: myRecords
+										backoff: err || 'error',
+										records: myRecords,
 									});
 								},
 								fail: (err) => {
 									done(null, {
-										fail: err || "error",
-										records: myRecords
-									})
+										fail: err || 'error',
+										records: myRecords,
+									});
 								},
 								success: () => {
 									done(null, {
 										success: true,
-										//records: myRecords
+										// records: myRecords
 									});
-								}
+								},
 							};
 							self.batchTableWrite(table, myRecords).then(unprocessedItems => {
 								if (unprocessedItems && Object.keys(unprocessedItems).length) {
-									myRecords = data.UnprocessedItems[table];
+									myRecords = self.data.UnprocessedItems[table];
```

Comment:
this should be like this

if (unprocessedItems && unprocessedItems.length) {
   myRecords = unprocessedItems;